### ISSUE TITLE
Chat SDK + Access Manager

### ISSUE THREAD

**Issue by manfe** [pubnub/js-chat#180]
When Access Manager is enabled I am initializing the Chat SDK with:

```js
const chatInstance = await Chat.init({
      publishKey: process.env.NEXT_PUBLIC_PUBNUB_PUBLISH_KEY,
      subscribeKey: process.env.NEXT_PUBLIC_PUBNUB_SUBSCRIBE_KEY,
      userId: "user-id-example",
      authKey: "my-token"
})
```

But if I need to update the access manager token after a period of time, I noticed that the pubnub is wrapped into the `sdk` property.

Is it ok to update the Access Manager token using the method below?

```js
chatInstance.sdk.setToken("my-new-token")
```

**Comment by piotr-suwala** [comment]
Hello @manfe!

Yes, it's okay to update the Access Manager token using setToken on the sdk property. However we noticed that a stale value will be used in the setLastReadMessageTimetoken and we need to fix it soon. You should be good to go with setToken.

Regarding your second question: you can use code like this if you use React:

```
  useEffect(() => {
    if (!chat) {
      return
    }

    const listeners = {
      status(statusEvent: Pubnub.StatusEvent) {
        if (
          statusEvent.category === "PNNetworkDownCategory" ||
          statusEvent.category === "PNNetworkIssuesCategory" ||
          statusEvent.category === "PNConnectionErrorCategory"
        ) {
        }
        // do something
      },
    }

    chat.sdk.addListener(listeners)

    return () => {
      chat.sdk.removeListener(listeners)
    }
  }, [chat])
```

**Comment by manfe** [comment]
@piotr-suwala another issue that I found.

If I have listeners to a channel or messages and the token is updated, those messages will keep the old token and while we don't stop listen and listen again (re-render all messages) with new listeners will not be able to toggle reaction on them or take any action with the stale token.

**Comment by piotr-suwala** [comment]
@manfe Did you use chatInstance.sdk.setToken("my-new-token") when your token is updated? Our PN Messages do not contain the AM token on their own.

**Comment by manfe** [comment]
@piotr-suwala yes, I was using this approach, but old messages couldn't have any kind o reaction.

So I had to initialize the chat again when a new token arrive.

**Comment by wkal-pubnub** [comment]
Hello, could you confirm if this is still a problem with version 0.9.6?
We've updated AccessManager to check for the current value of JS SDK's token

**Comment by jguz-pubnub** [comment]
Hi, @manfe. Invoking `chat.sdk.setToken("my-new-token")` when your token is updated is a valid approach. Additionally, it's possible to add connection status listeners starting from the `0.15.0` version, allowing you to listen to `403` errors as presented below. Could you please try it out?

```
chat.addConnectionStatusListener((connection) => {
    if (connection.exception?.status?.statusCode == 403) {
        // ...
    }
})
```

**Comment by manfe** [comment]
Hey, the project that I was working on was discontinued, so I can't re-check it anymore. I will close it. 

Thanks.
